internal/app/osd/internal/reg/reg.go

```diff
@@ -289,28 +284,27 @@ func (r *Registrator) Dmesg(ctx context.Context, in *empty.Empty) (data *proto.D
 // be requested and the contents of the log file are streamed in chunks.
 // nolint: gocyclo
 func (r *Registrator) Logs(req *proto.LogsRequest, l proto.OSD_LogsServer) (err error) {
-	var filename string
+	var chunk chunker.Chunker
 
 	switch {
 	case req.Namespace == "system" || req.Id == "kubelet" || req.Id == "kubeadm":
-		filename = filepath.Join("/var/log", filepath.Base(req.Id)+".log")
+		filename := filepath.Join("/var/log", filepath.Base(req.Id)+".log")
+		var file *os.File
+		file, err = os.OpenFile(filename, os.O_RDONLY, 0)
+		if err != nil {
+			return
+		}
+		// nolint: errcheck
+		defer file.Close()
```

Comment:
I believe this was missing in the previous implementation which should have lead to fd leak